controllers/extensions/addon_controller_stages.go

```diff
@@ -177,6 +178,19 @@ func (r *fetchNDeletionCheckStage) Handle(ctx context.Context) {
 	}
 	r.reqCtx.Log.V(1).Info("get addon", "generation", addon.Generation, "observedGeneration", addon.Status.ObservedGeneration)
 	r.reqCtx.UpdateCtxValue(operandValueKey, addon)
+
+	// CheckIfAddonUsedByCluster, if err, skip the deletion stage
+	if !addon.GetDeletionTimestamp().IsZero() || (addon.Spec.InstallSpec != nil && !addon.Spec.InstallSpec.GetEnabled()) {
```

Comment:
The check "addon.Spec.InstallSpec != nil" is unnecessary. since GetEnabled is a member function of pointer *AddonInstallSpec